### PR TITLE
Make opt-level=2 only in script crate tests and get rid of recursion to disconnect blocks

### PR DESCRIPTION
**Pull request by TheQuantumPhysicist**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -65,8 +65,11 @@ utxo = { path = "utxo"}
 [profile.release]
 debug-assertions = true
 overflow-checks = true
+lto = true
```
Comment (stanislav-tkach) [inline Cargo.toml:68]: Should we also use `codegen-units = 1`?

**Review by stanislav-tkach — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (stanislav-tkach) [reply Cargo.toml:68]: https://doc.rust-lang.org/rustc/codegen-options/index.html#codegen-units

File: Cargo.toml [hunk 2]
```diff
@@ -65,8 +65,11 @@ utxo = { path = "utxo"}
...
 
 [profile.test]
-opt-level = 2
 debug-assertions = true
 overflow-checks = true
+
+[profile.test.package.script]
+opt-level = 2
```
Comment (iljakuklic) [inline Cargo.toml:75]: Does this inherit the other properties of the "global" test profile?

**Review by iljakuklic — APPROVED** [review]
LGTM

File: Cargo.toml (hunk 2 above)
Comment (TheQuantumPhysicist) [reply Cargo.toml:75]: I think so. I only tested that script now is release.

File: Cargo.toml (hunk 1 above)
Comment (TheQuantumPhysicist) [reply Cargo.toml:68]: Yeah, I'm not sure... builds may become release slow...

File: Cargo.toml (hunk 2 above)
Comment (iljakuklic) [reply Cargo.toml:74]: Ok docs seem to suggest so, although it's not super explicit. Let's wait and see how long the CI takes.

File: Cargo.toml (hunk 1 above)
Comment (stanislav-tkach) [reply Cargo.toml]: OK, I didn't see that we're building everything in the release mode on CI.

**Review by muursh — APPROVED** [review]

**Comment by stanislav-tkach** [comment]
I found it really strange that enabling LTO breaks our code.

**Comment by TheQuantumPhysicist** [comment]
Related: https://github.com/mintlayer/mintlayer-core/issues/258

**Pull request merged**